### ISSUE TITLE
CMake install target not generating cmake config 

### ISSUE THREAD

**Issue by DraconPern** [SOCI/soci#394]
there's no way to include soci in another cmake project because the sociconfig.cmake file is missing from the installed directory.    Attempt to use FindSoci.cmake doesn't work either because the lib is decorated e.g. libsoci_core_3_2.lib

**Comment by DraconPern** [comment]
Basically, it doesn't work if one uses find_package in config mode nor module mode.

**Comment by mloskot** [comment]
@DraconPern TBH, I have not managed to wrap my head around the CMake project.config magic. Last time I tried I gave up on unclear documentation and lack of complete sample for a project that is more complex than basic 'hello world'.

CMake-experienced contributors strongly welcome :)

**Comment by DraconPern** [comment]
i'll work on some modifications to get it building on appveyor on windows.  I noticed that it's got switches for both shared and static build.  I think they should be mutually exclusive options?  boost does it that way, and most of the projects I have worked on also do the same.

**Comment by snikulov** [comment]
@DraconPern I prefer it should be exclusive and controlled by standard variable BUILD_SHARED_LIBS as pointed out here http://www.slideshare.net/DanielPfeifer1/cmake-48475415 slide 28 
@mloskot What do you think?

**Comment by mloskot** [comment]
@snikulov I agree. It was first time I read through Daniel Pfeifer's presentation. It's great resource, I wish I knew it before I created CMake setup for SOCI :)

**Comment by Krzmbrzl** [comment]
This should be resolved by #1118
